maven-plugin/src/main/java/org/openmrs/maven/plugins/BuildDistro.java

```diff
@@ -587,4 +592,85 @@ public boolean accept(File dir, String name) {
 			}
 		}
 	}
+
+	/**
+	 * Reads the content.properties file.
+	 * <p> For dependencies defined in both content.properties and distro.properties, this ensures the version specified in distro.properties 
+	 * matches the range defined in content.properties; otherwise, we output an error to the user to fix this.</p>
+	 * For dependencies not defined in distro.properties but defined in content.properties, the SDK should locate the latest matching version and add that to the OMODs and ESMs used by the distro.
+	 * 
+	 * @param userDir
+	 * @param distroProperties
+	 * @throws MojoExecutionException
+	 */
+	private void parseContentProperties(File userDir, DistroProperties distroProperties) throws MojoExecutionException {
+		File contentFile = new File(userDir, CONTENT_PROPERTIES);
+		if (!contentFile.exists()) {
+			log.info("No content.properties file found in the user directory.");
+			return;
+		}
+
+		Properties contentProperties = new Properties();
+		try (BufferedReader reader = new BufferedReader(new FileReader(contentFile))) {
+			contentProperties.load(reader);
+		} catch (IOException e) {
+			throw new MojoExecutionException("Failed to read content.properties file", e);
+		}
+
+		for (String dependency : contentProperties.stringPropertyNames()) {
+			String versionRange = contentProperties.getProperty(dependency);
+			String distroVersion = distroProperties.get(dependency);
+
+			if (distroVersion == null) {
+				String latestVersion = findLatestMatchingVersion(dependency, versionRange);
+				if (latestVersion == null) {
+					throw new MojoExecutionException("No matching version found for dependency " + dependency);
+				}
+
+				distroProperties.add(dependency, latestVersion);
+			} else {
+				checkVersionInRange(dependency, versionRange, distroVersion);
+			}
+		}
```

Comment:
Not all properties in the `content.properties` specify some sort of library. Specifically, I think it's only the keys that start: `omod.`, `owa.`, and `spa.frontendModule.` (plus, I suppose, `content.`

Reply:
Oh! and `war`!

Reply:
@ibacher For example, what will be the value for `content.`?

Reply:
`content.hiv=1.0.0` or something. If we have a content package that requires another content package.

Reply:
(Assume the default groupId is `org.openmrs.content`, but can be anything using the `content.hiv.groupId=xxx` pattern).

Reply:
ooh perfect!